### ISSUE TITLE
Support systemd-creds before machine-id is generated

### ISSUE THREAD

**Issue by Cornelicorn** [systemd/systemd#31895]
### Component

other

### Is your feature request related to a problem? Please describe

I just noticed you can't use systemd-creds without having a machine-id or a machine-id of `uninitialized` 
The encrypt command fails silently, even with `SYSTEMD_LOG_LEVEL`.
The setup and decrypt commands show the rather unhelpful messages
`probing initialization failed: No such file or directory` or `Failed to setup credentials host key: Package not installed` for a missing or uninitialized machine-id respectively.

### Describe the solution you'd like

A way to use systemd-creds on systems which don't have a machine-id set yet, e.g. when chrooting into a new install, so you can setup encrypted credentials for the first boot.

### Describe alternatives you've considered

Writing a machine-id, but that would make ConditionFirstBoot unusable.

### The systemd version you checked that didn't have the feature you are asking for

252 for the actual test, checked with the code on master

**Comment by Cornelicorn** [comment]
I don't know if this is really feasable or sensible for that matter.
If it's not, the documentation could state this and the error message should definitely be improved.

**Comment by poettering** [comment]
Which operation precisely doesn't work for you?

it appears you tried to encrypt a cred with the host key? That only works if the system *has* a host key, and that is linked to the machine id. 

Anyway, please explain in more detail what you are trying to do?

**Comment by Cornelicorn** [comment]
> Which operation precisely doesn't work for you?
> 
> it appears you tried to encrypt a cred with the host key? That only works if the system _has_ a host key, and that is linked to the machine id.
> 
> Anyway, please explain in more detail what you are trying to do?

Yes, I'm trying to encrypt with the host key from within a chroot. I don't want to generate a machine-id, since I want to use `ConditionFirstBoot=yes` for some services.
I saw that the host key is linked to the machine id in the code, yet it's not mentioned in the man pages as far as I can tell.

Lifting that restriction (I'm not sure what the implications would be) would solve this, otherwise documenting this somewhere would be nice.

**Comment by andsens** [comment]
> otherwise documenting this somewhere would be nice.

It *kind of* is, though barely: https://systemd.io/CREDENTIALS/#:~:text=Moreover%2C%20credentials%20cannot%20be%20prepared%20on%20a%20machine%20other%20than%20the%20local%20one.
> Moreover, credentials cannot be prepared on a machine other than the local one.
